addons/website_sale_mondialrelay/views/templates.xml

```diff
@@ -1,7 +1,7 @@
 <odoo>
 
     <template id="website_sale_mondialrelay_address_on_payment" inherit_id="website_sale.address_on_payment">
-        <xpath expr="//span[@t-esc='order.partner_shipping_id']" position="before">
+        <xpath expr="//span[@t-esc='order.partner_shipping_id'] | //span[@id='shipping_on_payment_details']" position="before">
```

Comment:
Isn't it another way to fix that? I'd rather avoid asking customers to update two modules.